services/api-server/src/simcore_service_api_server/api/routes/solvers_jobs.py

```diff
@@ -55,30 +62,21 @@ async def list_jobs(
     webserver_api: AuthSession = Depends(get_webserver_session),
     url_for: Callable = Depends(get_reverse_url_mapper),
 ):
-    """ List of all jobs in a specific released solver """
-
-    async def _fake_impl():
-        from ._jobs_faker import list_jobs_impl
+    """List of all jobs in a specific released solver"""
 
-        solver = await catalog_client.get_solver(user_id, solver_key, version)
-        return await list_jobs_impl(solver.id, solver.version, url_for)
+    solver = await catalog_client.get_solver(user_id, solver_key, version)
+    logger.debug("Listing Jobs in Solver '%s'", solver.name)
 
-    async def _draft_impl():
-        solver = await catalog_client.get_solver(user_id, solver_key, version)
-        logger.debug("Listing Jobs in Solver '%s'", solver.name)
+    projects: List[Project] = await webserver_api.list_projects(solver.name)
+    jobs: Deque[Job] = deque()
+    for prj in projects:
+        job = create_job_from_project(solver_key, version, prj, url_for)
+        assert job.id == prj.uuid  # nosec
```

Comment:
asserts?

Reply:
I use asserts as a mechanism to check "conditions given by design" while developing . Some people call this *defensive programming*. I think it is a nice way to document+test while development. 
In production, these asserts are all disabled (see the ``PYTHONOPTIMIZE`` variable in ``Dockerfile`` ) so you they are not part of the workflow and will never raise.